master/buildbot/changes/gerritchangesource.py

```diff
@@ -29,6 +30,33 @@
 from buildbot.util import httpclientservice
 
 
+def canonicalize_event(event):
+    """
+    Return an event dictionary which is consistent between the gerrit
+    event stream and the gerrit event log formats.
+    """
+    # For "patchset-created" the events-log JSON looks like:
+    #   "project": {"name": "buildbot"}
+    # while the stream-events JSON looks like:
+    #   "project": "buildbot"
+    # so we canonicalize them to the latter
+    if ("change" in event and "project" in event["change"] and
```

Comment:
what bothers me a bit is a multiple lookups of the same key... it's probably not a problem in a production environment, however it's not a good style in general...

Reply:
I disagree regarding style but I don't really care one way or the other. Done.